From: Michael Karcher <debian@mkarcher.dialup.fu-berlin.de>
Date: Tue, 2 Jul 2013 11:02:23 +0200
Subject: Fix_LP_#911146

Ensures that filenames passed to extensions are made absolute
before changing the current working directory.

Signed-off-by: Matteo F. Vescovi <mfv.debian@gmail.com>
---
 src/extension/implementation/script.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/extension/implementation/script.cpp b/src/extension/implementation/script.cpp
index 5c485a6..b4e9847 100644
--- a/src/extension/implementation/script.cpp
+++ b/src/extension/implementation/script.cpp
@@ -956,7 +956,14 @@ int Script::execute (const std::list<std::string> &in_command,
     // assemble the rest of argv
     std::copy(in_params.begin(), in_params.end(), std::back_inserter(argv));
     if (!filein.empty()) {
-        argv.push_back(filein);
+        if(Glib::path_is_absolute(filein))
+            argv.push_back(filein);
+        else {
+            std::vector<std::string> buildargs;
+            buildargs.push_back(Glib::get_current_dir());
+            buildargs.push_back(filein);
+            argv.push_back(Glib::build_filename(buildargs));
+        }
     }
 
     int stdout_pipe, stderr_pipe;
